### PR TITLE
Add GetFormatPatch receiver for a repository

### PR DESCRIPTION
**Pull request by strk**
Includes test

### CODE REVIEW

**Comment by strk** [comment]
I don't know why `GetPatch` returns a `byte[]`, for simplicity I had `GetFormatPatch` return a `string` instead

**Comment by lafriks** [comment]
Could diff contain binary data?

**Comment by strk** [comment]
I guess so. Ok will add --binary switch and return binary. Will need to find out how to convert the `byte[]` to a string for comparison against Regexp in test

**Comment by strk** [comment]
pushed binary output, and the test file which was previously missing

**Comment by lafriks** [comment]
Also I think it would be better to use RunInDirPipeline and stream stdout io.Copy it later directly to response writer

**Comment by strk** [comment]
yes, I like that better too, now if I'd knew how to do it :)
Same for GetDiff ... but we need a different name for the piping version, so not to break API

File: repo_pull_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package git
+
+import (
+	"testing"
```
Comment (appleboy) [inline repo_pull_test.go]: maybe add an empty line here.

Comment (strk) [reply repo_pull_test.go]: done with 9783381

File: repo_pull.go [hunk 2]
```diff
@@ -73,3 +73,8 @@ func (repo *Repository) GetPullRequestInfo(basePath, baseBranch, headBranch stri
 func (repo *Repository) GetPatch(base, head string) ([]byte, error) {
 	return NewCommand("diff", "-p", "--binary", base, head).RunInDirBytes(repo.Path)
 }
+
+// GetFormatPatch generates and returns format-patch data between given revisions.
+func (repo *Repository) GetFormatPatch(base, head string) ([]byte, error) {
```
Comment (bkcsoft) [inline repo_pull.go]: For the gitea PR I'd implement a `GetFormatPatchReader(base, head string) (io.ReadCloser, error)` as well.

Comment (strk) [reply repo_pull.go]: I'm out of time for today so what about merging this and then improve later ? I'm looking forward for your PR implementing the "Reader" version of both methods (GetPatchReader and GEtFormatPatchReader)

Comment (robe2) [reply repo_pull.go]: Merging @strk  PR would be much appreciated.

**Pull request merged**